docs/guide/cursors.txt

```diff
@@ -1,3 +1,198 @@
 =======
 Cursors
 =======
+
+When a driver executes a read that returns mutliple documents, the server does not immediately return all values that match the query. Instead, the driver creates a **Cursor** object which fetches the documents in batches.
+
+Where are cursors used in the driver?
+-------------------------------------
+
+The following functions directly return cursors to the user:
+
+
+* ``Collection.prototype.find``
+* ``Collection.prototype.aggregate``
+* ``Collection.prototype.listIndexes``
+* ``Db.prototype.aggregate``
+* ``Db.prototype.listCollections``
+
+In addition, many other methods, like ``Collection.prototype.findOne`` and ``Collection.prototype.watch``\ , use cursors in order to return results.
+
+How do I use a cursor?
+----------------------
+
+There are multiple ways to consume a cursor:
+
+Stream API
+^^^^^^^^^^
+
+All cursors in the Node Driver are Node Readable Streams operating in Object Mode. Cursors will work with most Node stream APIs.
+
+.. code-block:: js
+
+   // defines a transform that calls JSON.stringify on the cursor's output
+   class StringifyStream extends stream.Transform {
+     constructor() {
+       // Need this b/c cursors operate in objectMode
+       super({ objectMode: true });
+     }
+
+     transform(chunk, _, callback) {
+       // outputs each document as a stringified version, separated by newlines
+       callback(null, `${JSON.stringify(chunk)}\n`);
+     }
+   }
+
+   // Using the stream API to pipe to stdout
+   const cursor1 = collection.find({});
+   cursor1
+     .pipe(new StringifyStream())
+     .pipe(process.stdout);
+   
+   // Using the stream API to pipe to file 'my-output.txt'
+   const cursor2 = collection.find({});
+   cursor2
+     .pipe(new StringifyStream())
+     .pipe(fs.createWriteStream('my-output.txt'));
+
+Event API
+^^^^^^^^^
+
+As Readable Streams, Cursors also support an Event API.
+
+.. code-block:: js
+
+   // Using the event API
+   const cursor = collection.find({});
+   cursor.on('data', data => console.log(data));
+
+Get all documents at once
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To get all documents at once, users can use the ``toArray`` method.
+
+.. code-block:: js
+
+   // Get all values
+   const cursor = collection.find({});
+   const allValues = await cursor.toArray();
+
+Async Iterator
+^^^^^^^^^^^^^^
+
+Cursors also implement the AsyncIterator interface, allowing them to be used in ``for``...\ ``await`` loops.
+
+.. code-block:: js
+
+   const cursor = collection.find({});
+   for await(const doc of cursor) {
+     console.log(doc);
+   }
+
+For Each
+^^^^^^^^
+
+If you want to perform an operation on each document, you can use the ``forEach`` method.
+
+.. code-block:: js
+
+   // apply a callback to every item in a cursor
+   const cursor = collection.find({});
+   await cursor.forEach(doc => console.log(doc));
+
+Get a single document
```

Comment:
I think the content of this section describes what we might call a "Java-style iterator API", rather than "Get a single document". It's introducing `hasNext` and `next` and showing how to iterate using them. I think we can leave it up to the reader's imagination how to just get a single document with this API if that's what they want